l10n/src/main.ts

```diff
@@ -151,6 +176,18 @@ export function t(...args: [str: string, ...args: Array<string | number | boolea
         message = firstArg;
         args.splice(0, 1);
         formatArgs = !args || typeof args[0] !== 'object' ? args : args[0];
+    } else if (firstArg instanceof Array) {
+        const replacements = args.slice(1) as L10nReplacement[];
+        if (firstArg.length !== replacements.length + 1) {
+            throw new Error('expected a string as the first argument to l10n.t');
```

Comment:
yes, technically the first argument can be an array for template literals, but this code path would be hit by someone calling the standard `l10n.t()` method and passing something bad as the first argument